pkg/controllers/cluster/cluster_controller.go

```diff
@@ -240,6 +241,9 @@ func (c *Controller) syncCluster(ctx context.Context, cluster *clusterv1alpha1.C
 
 func (c *Controller) removeCluster(ctx context.Context, cluster *clusterv1alpha1.Cluster) (controllerruntime.Result, error) {
 	// add terminating taint before cluster is deleted
+	defer func() {
+		metrics.RemoveClusterStatus(cluster)
+	}()
```

Comment:
What if the removeCluster raises an error? Do we still clean up the metrics?

Reply:
> What if the removeCluster raises an error? Do we still clean up the metrics?

shouldn't clean up the metrics, I'll fix it!

Reply:
> Have you tested it? Can you share the metrics before and after cluster is deleted?
I join a cluster named `host`
Before metrics:  
```
cluster_ready_node_number{cluster_name="host"} 0
cluster_ready_state{cluster_name="host"} 1
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.005"} 0
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.01"} 0
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.025"} 0
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.05"} 1
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.1"} 3
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.25"} 3
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="0.5"} 4
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="1"} 4
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="2.5"} 4
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="5"} 4
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="10"} 4
cluster_sync_status_duration_seconds_bucket{cluster_name="host",le="+Inf"} 4
cluster_sync_status_duration_seconds_sum{cluster_name="host"} 0.667916613
cluster_sync_status_duration_seconds_count{cluster_name="host"} 4
```